### PR TITLE
feat: implement `Open()` on `LookupEvaluated`

### PR DESCRIPTION
**Pull request by lightscale-luke**
* revert using `ref_aliases.h`
* add move constructors and unit tests on `ProverQuery` and `VerifierQuery`
* implement `Open()` on `LookupEvaluated`

### CODE REVIEW

**Comment by chokobole** [comment]
Please change "build(zk): tidy up bazel build" to "refac(zk): tidy up headers"

**Comment by chokobole** [comment]
Please include 6dd577c4675d6f08dcd5bb1139f94653c08f242c in "feat(zk): revert using ref_aliases.h" for reference.

File: tachyon/zk/base/prover_query_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+}  // namespace
+
+TEST_F(ProverQueryTest, Constructor) {
+  ProverQuery<PCS> prover_query(point_, blinded_poly_ref_);
+
+  EXPECT_EQ(prover_query.point(), point_);
+  EXPECT_EQ(prover_query.poly()->poly(), blinded_poly_.poly());
+  EXPECT_EQ(prover_query.poly()->blind(), blinded_poly_.blind());
```
Comment (chokobole) [inline tachyon/zk/base/prover_query_unittest.cc:32]:
```suggestion
  EXPECT_EQ(prover_query.poly(), blinded_poly_);
```

File: tachyon/zk/base/verifier_query_unittest.cc
Comment (chokobole) [inline tachyon/zk/base/verifier_query_unittest.cc:1]: I don't think this kind of unittests are worth doing it.

File: tachyon/zk/base/prover_query_unittest.cc
Comment (chokobole) [inline tachyon/zk/base/prover_query_unittest.cc:1]: I don't think this kind of unittests are worth doing it.

**Review by chokobole — CHANGES_REQUESTED** [review]

File: tachyon/zk/base/prover_query_unittest.cc (hunk 1 above)
Comment (lightscale-luke) [reply tachyon/zk/base/prover_query_unittest.cc:32]: unit test deleted

File: tachyon/zk/base/prover_query_unittest.cc
Comment (lightscale-luke) [reply tachyon/zk/base/prover_query_unittest.cc:1]: ok!. i will delete the unit tests

File: tachyon/zk/base/verifier_query_unittest.cc
Comment (lightscale-luke) [reply tachyon/zk/base/verifier_query_unittest.cc:1]: ok!. i will delete the unit tests

File: tachyon/zk/plonk/lookup/lookup_evaluated.h [hunk 2]
```diff
@@ -35,6 +37,19 @@ class LookupEvaluated {
   }
   const BlindedPolynomial<Poly>& product_poly() const { return product_poly_; }
 
+  template <typename ProverTy>
```
Comment (TomTaehoonKim) [inline tachyon/zk/plonk/lookup/lookup_evaluated.h:40]:
```suggestion
  template <typename ProverTy, typename PCSTy = ProverTy::PCSTy>
```

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by chokobole — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by Insun35 — COMMENTED** [review]
Btw, the CI has failed. Please check.

**Pull request merged**